src/test/java/org/folio/orders/events/handlers/ReceiptStatusConsistencyTest.java

```diff
@@ -0,0 +1,147 @@
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.instanceOf;
+
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.rest.jaxrs.model.PoLine.ReceiptStatus;
+import org.folio.rest.acq.model.PoLine;
+import org.folio.rest.acq.model.Piece.ReceivingStatus;
+import org.folio.rest.impl.ApiTestBase;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.core.Response;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.eventbus.ReplyException;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class ReceiptStatusConsistencyTest extends ApiTestBase {
+  private static final Logger logger = LoggerFactory.getLogger(ReceiptStatusConsistencyTest.class);
+
+  public static final String TEST_ADDRESS = "testReceiptStatusAddress";
+  private static final String VALID_PIECE_ID = "af372ac8-5ffb-4560-8b96-3945a12e121b";
+  private static final String PIECE_ID_WITH_NO_PO_LINE = "5b454292-6aaa-474f-9510-b59a564e0c8d";
+  private static final String PIECE_ID_WITH_500_ON_PO_LINE = "7d0aa803-a659-49f0-8a95-968f277c87d7";
+  private static final String EXPECTED = "Expected";
+  private static final String RECEIVED = "Received";
+
+  private static Vertx vertx;
+
+  @BeforeClass
+  public static void before() throws InterruptedException, ExecutionException, TimeoutException {
+    ApiTestBase.before();
+
+    vertx = Vertx.vertx();
+    vertx.eventBus().consumer(TEST_ADDRESS, new ReceiptStatusConsistency(vertx));
+  }
+
+  @Test
+  public void testReceiptStatusWhenReceiptStatusBetweenPiecesAndPoLineNotConsistent(TestContext context) {
+    logger.info("=== Test case when piece receipt status changes from Received to Expected ===");
+    
+    // explicitly setting RECEIVED to create inconsistency
+    sendEvent(createBody(RECEIVED, VALID_PIECE_ID), context.asyncAssertSuccess(result -> {
+      logger.info("getPoLineSearches()--->" + getPoLineSearches());
+      logger.info("getPoLineUpdates()--->" + getPoLineUpdates());
+      logger.info("getPieceSearches()--->" + getPieceSearches());
+      assertEquals(2, getPieceSearches().get(0).getJsonArray("pieces").size());
+      
+      Piece piece0 = getPieceSearches().get(0).getJsonArray("pieces").getJsonObject(0).mapTo(Piece.class);
+      Piece piece1 = getPieceSearches().get(0).getJsonArray("pieces").getJsonObject(1).mapTo(Piece.class);
+      assertEquals(piece0.getReceivingStatus().toString().toLowerCase(),ReceivingStatus.EXPECTED.value().toLowerCase());
```

Comment:
no need for the toString/toLowerCase stuff... just compare the enum values.

Reply:
Done